### PR TITLE
Add detailes about Segment sending logic.

### PR DESCRIPTION
**Pull request by ifireball**
Focus on how to make this logic reliable and observable while also require very little code to implement.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,3 +114,48 @@ The interesting fields for this use case are:
   the SSO user ID to be sent to Segment
 - `status.compliantUsername` - Contains the username used in the cluster audit
   log.
+
+## Details about the Event Sender
+
+Segment has a [built-in mechanism for removing duplicate events][ES1]. This
+mean that we can safely resend the same event multiple times to increase the
+sending process reliability. The duplicate remove mechanism is based on the
+`messageid` [common message field][ES2]. We can the `auditID` field of the
```
Comment (Omeramsc) [inline README.md:123]:
```suggestion
`messageid` [common message field][ES2]. We can use the `auditID` field of the
```

**Review by Omeramsc — CHANGES_REQUESTED** [review]
Overall LGTM, sending data with duplication prevention in mind sounds good. any plans on how will we test this?

**Comment by ifireball** [comment]
> Overall LGTM, sending data with duplication prevention in mind sounds good. any plans on how will we test this?

I guess we can look for duplicates in Amplitude once we can some data in there, but with buth Segment and Amplitude being proprietary SAAS systems, there is no much testing we can do around them.

**Review by eisraeli — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -114,3 +114,48 @@ The interesting fields for this use case are:
...
+given conditions. The [transaction][ES5] command allows for the grouping of
+multiple search records into a single record with multi-value fields while the
+`mv_to_json_array` [eval][ES6] function lets one convert such fields into JSON
+lists. Along with the [json_object][ES7] function these could be used to
+convert Splunk query results into a series of JSON objects where each does not
+exceed a given size.
+
+The architecture for the event sending logic would then be:
+
+1. Repeat the following logic on an hourly basis:
```
Comment (eisraeli) [inline README.md:142]: You might want to remove the `1.` here. Looks unnecessary.

Comment (ifireball) [reply README.md]: done.

**Review by Omeramsc — APPROVED** [review]
LGTM

**Pull request merged**